projects/angular-query/src/lib/angular-query.utils.ts

```diff
@@ -0,0 +1,40 @@
+import type { MutationAction, MutationState } from './angular-query.types';
+import { getStatusProps } from './query/core/utils';
+
+export const getDefaultMutationState = <TData, TError>(): MutationState<TData, TError> => {
+  return {
+    ...getStatusProps('idle'),
+    data: undefined,
+    error: null,
+  };
+};
+
+export const reducer = <TData, TError>(
```

Comment:
What do you think of being a little more concise in our naming?
```suggestion
export const mutationReducer = <TData, TError>(
```
When I saw it in the other file I tought it was the `reducer` operator from RxJS!
mutationReducer perhaps? This could be applied elsewhere but I didn't want to nitpick everywher :)

I was wondering if you skipped that part of the code but you didn't 💪

Reply:
Yes, we can reuse this [code](https://github.com/tannerlinsley/react-query/blob/master/src/react/useMutation.ts#L72)